nomad/core_sched.go

```diff
@@ -241,16 +241,18 @@ func (c *CoreScheduler) gcEval(eval *structs.Evaluation, thresholdIndex uint64,
 	// Create a watchset
 	ws := memdb.NewWatchSet()
 
+	// Look up the job
+	job, err := c.snap.JobByID(ws, eval.Namespace, eval.JobID)
```

Comment:
Will c.snap ever be nil?

Reply:
It is set in the constructor by the worker, should be non nil.